Fix helpdesk email links

ericamador opened the pull request.

ericamador committed:
Pull all imports to the top of models.py

This makes it easier to see everything that this file is importing.

ericamador committed:
Remove redundant import from lib.py

This import already exists in the one function where it is used (it
exists there to prevent a ciruclar dependency).

ericamador committed:
Move Attachment model import in lib.py to the function where it is used

This is done to avoid a circular dependecy with models.py

ericamador committed:
Pull all imports to top of lib.py

This is to make it easier to see what is being imported by this file.

ericamador committed:
Remove references to text_is_spam function

This is dead code that is never used anywhere.

ericamador committed:
Fix line lengths in lib.py to comply with flake8

This is done for style purposes.

ericamador committed:
Fix long lines in models.py

This commit fixes the line lengths in models.py to comply with flake8.

ericamador committed:
Remove complexity from _aboslute_uri in models.py

Previously, the code would first try to build a URL using the request object
if one was available. However, the "if self.request" was often raising
an uncaught AttributeError because the request didn't exist. We address
this problem by simply always using the current Site to build the URL.
This has the added benefit of reducing the cyclomatic complexity of this
function, making the behavior easier to reason about and maintain.

ericamador committed:
Update CHANGES and setup.py for new release v0.2.6+nimbis.15

ericamador commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Fix broken email links in updated/closed django-helpdesk ticket emails](https://trello.com/c/VI7Ihwhj/2806-fix-broken-email-links-in-updated-closed-django-helpdesk-ticket-emails)

cworth-gh commented on helpdesk/lib.py:
```diff
@@ -274,8 +275,6 @@ def text_is_spam(text, request):
     # False if it is not spam. If it cannot be checked for some reason, we
     # assume it isn't spam.
-    from django.contrib.sites.models import Site
-    from django.core.exceptions import ImproperlyConfigured
```
I see this commit removing these two imports but not also adding them above.

Have you checked the code with flake8?

ericamador replied:
Yes, these imports were only used the the dead Akismet function that was removed.

cworth-gh commented on helpdesk/models.py:
```diff
@@ -577,19 +577,13 @@ def _get_status(self):
+        try:
+            site = Site.objects.get_current()
+        except:
+            site = Site(domain='configure-django-sites.com')
```
"configure-django-sites.com", really? Is that just an attempt to embed a little bit of documentation into the eventual error message? Like: "Site not found: configure-django-sites.com"... leads to "Oh, I guess I'd better configure django sites!" ??

Admittedly, I've been known to do my fair share of this kind of thing, (in code where I can't control the error message directly, but I'd like to put some information in it).

But at the very least, I would have put in a good comment explaining what was going on.

Anyway, I know this wasn't our code originally. But I also wonder if we wouldn't want to have an actual failure here, (so that _we_ get an actual error instead of the user silently getting a bogus domain).

ericamador replied:
I'll change this to raise an exception instead. Thanks for the feedback!

ericamador committed:
Make URL generation raise exception rather than use default Site

This will make the need to ensure that a Django Site object exists much
more obvious.
